starlette/testclient.py

```diff
@@ -455,11 +455,12 @@ def __enter__(self) -> requests.Session:
         self.receive_queue = asyncio.Queue()  # type: asyncio.Queue
         self.task = loop.create_task(self.lifespan())
         loop.run_until_complete(self.wait_startup())
-        return self
+        return super().__enter__()
 
     def __exit__(self, *args: typing.Any) -> None:
         loop = asyncio.get_event_loop()
```

Comment:
I think this is causing the problem with the loop not closing:

I think a loop context that allows intermediate blocking IO might be best as this:

```python
import asyncio.runners
improt contextlib

def _get_running_loop(*args, **kwargs):
    try:
        return asyncio.get_running_loop()
    except RuntimeError:
        return None

@contextlib.contextmanager
def new_loop():
    if _get_running_loop() is not None:
        raise RuntimeError(
            "new_loop() cannot be called from a running event loop"
        )

    with exitstack.ExitStack() as stack:
        loop = events.new_event_loop()
        stack.callback(loop.close)

        asyncio.set_event_loop(loop)
        stack.callback(asyncio.set_event_loop, None)

        @stack.callback
        def _shutdown_asyncgens():
            loop.run_until_complete(loop.shutdown_asyncgens())

        stack.callback(asyncio.runners._cancel_all_tasks, loop)
        yield loop
```